### ISSUE TITLE
Solve weird sqlite error

### ISSUE THREAD

**Issue by matz-e** [matz-e/lobster#439]
See discussion in 937c751fa392ced54b41f189bdefcc21bc153442, and from slack:
```
Traceback (most recent call last):
  File "/afs/crc.nd.edu/user/m/mbeydler/.lobster/bin/lobster", line 9, in <module>
    load_entry_point('Lobster==1.5', 'console_scripts', 'lobster')()
  File "/afs/crc.nd.edu/user/m/mbeydler/.lobster/lib/python2.7/site-packages/Lobster-1.5-py2.7.egg/lobster/ui.py", line 109, in boil
    args.plugin.run(args)
  File "/afs/crc.nd.edu/user/m/mbeydler/.lobster/lib/python2.7/site-packages/Lobster-1.5-py2.7.egg/lobster/./commands/process.py", line 159, in run
    self.sprint()
  File "/afs/crc.nd.edu/user/m/mbeydler/.lobster/lib/python2.7/site-packages/Lobster-1.5-py2.7.egg/lobster/./commands/process.py", line 174, in sprint
    task_src = TaskProvider(self.config)
  File "/afs/crc.nd.edu/user/m/mbeydler/.lobster/lib/python2.7/site-packages/Lobster-1.5-py2.7.egg/lobster/core/source.py", line 193, in __init__
    self.__store.register_dataset(wflow, dataset_info, wflow.category.runtime)
  File "/afs/crc.nd.edu/user/m/mbeydler/.lobster/lib/python2.7/site-packages/Lobster-1.5-py2.7.egg/lobster/core/unit.py", line 209, in register_dataset
    dataset_info.total_events))
sqlite3.OperationalError: no such table: workflows
```

Seems like ELKs `check_client` may in some weird ways interfere with things?

**Comment by klannon** [comment]
Three suggestions from @btovar and @dthain:

 1.  Turn off classic plotting or otherwise prevent that second thread from being created to ensure there is only (at least as far as we're concerned) one thread accessing the DB.  See if that changes the error at all.
 2.  Add a sleep statement immediately before the statement that fails (I think this would be the `register_dataset` call) and see if the statement succeeds if we sleep for a while.
 3.  Immediately after creating the Lobster DB, change the access permissions so that no one (including you) can read it.  This will cause anything accessing the DB to complain, which  might give us some hint about what other access might be causing this issue.

Note:  None of these suggestions actually fixes the problem, but they would hopefully give more information about what's causing it.

**Comment by annawoodard** [comment]
Running on 71f0629:
With automatic plotting and ELK enabled: got the error
With automatic plotting and ELK disabled: got the error
With automatic plotting disabled and ELK disabled: got the error

Note that if I disable the DBS check in `cmssw.Dataset.validate()` I don't get the error:
```
      def validate(self):
          if self.dataset in Dataset.__dsets:
              return True

          if self.lumi_mask:
              self.lumi_mask = self.__get_mask(self.lumi_mask)

          # if self.dbs_instance not in self.__apis:
          #     cred = Proxy({'logger': logging.getLogger("WMCore")})
          #     dbs_url = 'https://cmsweb.cern.ch/dbs/prod/{0}/DBSReader'.format(self.dbs_instance)
          #     self.__apis[self.dbs_instance] = DASWrapper(dbs_url, ca_info=cred.getProxyFilename())
 
          # baseinfo = self.__apis[self.dbs_instance].listFileSummaries(dataset=self.dataset)
          # if baseinfo is None or (len(baseinfo) == 1 and baseinfo[0] is None):
          #     return False
          return True
```

**Comment by annawoodard** [comment]
Running on 71f0629 with automatic plotting disabled and ELK disabled and sleeping for 30 seconds before `register_datasets`: same error.

**Comment by annawoodard** [comment]
Deleting the `DbsApi` object after using it solves the problem. It's not that expensive to make a new one each time we use it (which is only at the beginning of runs)-- I vote we do that. Any objections?

**Comment by btovar** [comment]
Yes?

What is about deleting the object that fixes the problem?

**Comment by annawoodard** [comment]
I am not sure yet. Immediately changing the permissions of the Lobster DB to `000` as Ben and Doug suggested also solves the problem (but obviously breaks plotting, etc, because subsequent connections can't read the DB and fail with `sqlite3.OperationalError: unable to open database file`). Unfortunately though, I don't get any other errors (for example, from DBS) that would give us a hint about what other access is causing the issue. The master fires up and runs fine.

**Comment by matz-e** [comment]
Good catch! With the error we got before with the ELK branch, that makes me think that we shouldn't have any persistent network connections before we daemonize…

**Comment by annawoodard** [comment]
@matze just to clarify: I see the same error when running in foreground mode.

**Comment by matze** [comment]
@annawoodard: the difference between your intended recipient (@matz-e) and me (@matze) is a dash between *z* and *e*. I brought that up in another issue already …

**Comment by matz-e** [comment]
(That is probably because I normally don't have the dash in my username in other systems)

**Comment by annawoodard** [comment]
@matze: Sorry about that. It came up because I was writing on mobile and my keyboard autocorrected to matze. I'll delete it from the dictionary and be more careful.

**Comment by annawoodard** [comment]
> I'm wondering what happens if we drop caching the DBS API completely? It should be pretty cheap to re-create it for every dataset, given that querying the database is way more expensive…

@matz-e: To clarify, that's what I tested and was suggesting above. I believe the caching is a vestige from the old days anyways, and doesn't make much sense in the current implementation (I don't believe it can ever be useful because each `Workflow` has its own `Dataset`.)

It's still not clear to me exactly what's causing the error (we can't answer @btovar's question precisely). In any case, we've got physics to do, so I agree we should just move forward with this solution.
